### PR TITLE
Refactoring  Cauldrons page

### PR DESCRIPTION
**Pull request by Andrey-Onipchenko**

### CODE REVIEW

File: src/views/markets/Cauldrons.vue [hunk 1]
```diff
@@ -1,432 +1,256 @@
...
-      v-if="showButtonUp"
-      alt=""
-    />
+  <div class="cauldrons-page">
     <h2 class="title">Available Cauldrons</h2>
-
-    <EmptyMarketsList v-if="!borrowPools.length && !borrowLoading" />
-    <div v-else-if="!borrowPools.length && borrowLoading" class="loader-wrap">
+    <EmptyState v-if="isEmptyState" />
+    <div v-else-if="cauldronsIsLoaded" class="loader-wrap">
```
Comment (0xhardrefresh) [inline src/views/markets/Cauldrons.vue:5]: isLoading

File: src/components/ui/dropdown/SortBy.vue [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        alt="Sort reverse"
+      />
+      <span>{{ `Sorted by ${dropdownTitle}` }}</span>
+      <img
+        class="arrow-icon"
+        src="@/assets/images/arrow-down.svg"
+        alt="Arrow"
+      />
+    </button>
+    <div class="dropdown-list" v-show="isOpenDropdawn">
```
Comment (0xidkfa) [inline src/components/ui/dropdown/SortBy.vue:22]: Typo. Should be `isOpenDropdown`.

File: src/components/ui/dropdown/SortBy.vue [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      <button
+        class="dropdown-item"
+        v-for="(titleData, i) in dropdownList"
+        @click="changeDropdawnValue(titleData.name)"
```
Comment (0xidkfa) [inline src/components/ui/dropdown/SortBy.vue:26]: Typo. Should be `changeDropdownValue`.

File: src/helpers/cauldron/position/getUserCollateralInfo.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  userCollateralAmount: BigNumber;
+};
+
+export const getUserCollateralInfo = async (
+  user: string,
+  bentoBox: Contract,
+  cauldron: Contract,
+  collateral: string
+): Promise<UserCollateralInfo> => {
+  const userCollateralShare: BigNumber = await cauldron.userCollateralShare(
```
Comment (0xidkfa) [inline src/helpers/cauldron/position/getUserCollateralInfo.ts:14]: Probably better to use lens here...

File: src/helpers/cauldron/position/getUserLiquidationPrice.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+export const getLiquidationPrice = (
+  userCollateralShare: string | number,
+  userBorrowPart: string | number,
+  mcr: number
+): number => {
+  if(Number(userCollateralShare) === 0) return 0;
+  if(Number(userBorrowPart) === 0) return 0;
+
+  const liquidationPrice =
```
Comment (0xidkfa) [inline src/helpers/cauldron/position/getUserLiquidationPrice.ts:9]: Lens?

File: src/components/markets/CauldronItem.vue [hunk 6]
```diff
@@ -36,94 +47,72 @@ import BaseTokenIcon from "@/components/base/BaseTokenIcon.vue";
...
-      if (this.chainId === 250) {
-        return this.$image("assets/images/networks/fantom-icon.svg");
-      }
-
-      if (this.chainId === 43114) {
-        return this.$image("assets/images/networks/avalanche-icon.png");
-      }
-
-      if (this.chainId === 137) {
-        return this.$image("assets/images/networks/polygon-icon.svg");
```
Comment (0xCalibur) [inline src/components/markets/CauldronItem.vue:93]: great!

File: src/components/ui/dropdown/SortBy.vue (hunk 2 above)
Comment (Andrey-Onipchenko) [reply src/components/ui/dropdown/SortBy.vue:22]: fix

File: src/components/ui/dropdown/SortBy.vue (hunk 3 above)
Comment (Andrey-Onipchenko) [reply src/components/ui/dropdown/SortBy.vue:26]: fix

File: src/components/ui/search/CauldronsSearch.vue [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      class="search-input"
+      @input="changeSerch"
+    />
+  </div>
+</template>
+
+<script>
+export default {
+  methods: {
+    changeSerch(event) {
```
[the remark this answers is not in the document]
Comment (Andrey-Onipchenko) [reply src/components/ui/search/CauldronsSearch.vue:20]: updated

**Review by 0xidkfa — APPROVED** [review]

File: src/components/markets/EmptyState.test.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+describe("EmptyState.vue", () => {
+  it("Should render default string without props passed", () => {
+    const wrapper = mount(EmptyState);
+    const stateText = wrapper.find("p.state-text");
+    expect(stateText.text()).toBe(
+      "No pools on this network. In the future they will be displayed here"
+    );
+  });
+
+  it("Should render default string without props passed", () => {
```
[the remark this answers is not in the document]
Comment (Andrey-Onipchenko) [reply src/components/markets/EmptyState.test.ts]: updated

**Review by 0xCalibur — APPROVED** [review]

**Pull request merged**